### PR TITLE
refactor terraform plan runner

### PR DESCRIPTION
**Pull request by YaaraVerner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

### Description
refactor terraform_plan runner - split the logic to the runner and plan_utils.
create plan_utils that include calculating the definitions and building definitions context.
add an option to inject the definitions, the context, and the graph manager in order to not calculate it in the run function.
run the checks on the tf_definitions after calculating all of them.
build the graph with all the files data and not only the last.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/plan_parser.py [hunk 1]
```diff
@@ -125,15 +125,13 @@ def _find_child_modules(child_modules: ListNode) -> List[Dict[str, Dict[str, Any
     return resource_blocks
 
 
-def parse_tf_plan(tf_plan_file: str, out_parsing_errors: Dict[str, str]) -> Tuple[Optional[Dict[str, Dict[str, Any]]], Optional[List[Tuple[int, str]]]]:
+def parse_tf_plan(tf_defintions: Dict[str, Dict[str, Any]], template_lines: Dict[str, List[Tuple[int, str]]], tf_plan_file: str, out_parsing_errors: Dict[str, str]):
```
Comment (tronxd) [inline checkov/terraform/plan_parser.py:128]: @YaaraVerner why does this method now accepts `tf_definitions` and not returning it? `create_definitions` anyways return the definitions, so I don't see a reason to add additional arguments, and It's not a best practice to modify in-place a function argument, unless we have a good reason for it

Comment (YaaraVerner) [reply checkov/terraform/plan_parser.py:128]: until now for each plan file, we parsed it and run the checks on it. the function is parsing only one file and I wanted to save all the results in the same object. but I agree that it isn't the best solution, I will change it and extend the tf_definitions outside the function.

Comment (YaaraVerner) [reply checkov/terraform/plan_parser.py]: @tronxd done :)

**Review by rotemavni — APPROVED** [review]
🥇

**Pull request merged**